upcloud/resource_upcloud_objectstorage_test.go

```diff
@@ -349,16 +349,20 @@ func verifyObjectStorageExists(accessKey, secretKey, name string) resource.TestC
 
 func verifyObjectStorageDoesNotExist(accessKey, secretKey, name string) resource.TestCheckFunc {
 	return func(state *terraform.State) error {
-		time.Sleep(time.Second * 3)
-		exists, err := doesObjectStorageExists(state, accessKey, secretKey)
-		if err != nil {
-			if err.Error() == "could not find resources" {
-				return nil
+
+		for _, rs := range state.RootModule().Resources {
+			if rs.Type != "upcloud_storage" {
+				continue
+			}
+
+			client := testAccProvider.Meta().(*service.Service)
+			_, err := client.GetObjectStorageDetails(&request.GetObjectStorageDetailsRequest{
```

Comment:
Is there a reason why we're using a 'custom' way to detect this instead of the helper function `doesObjectStorageExists`? If there is, we probably should get rid of the helper function and roll it into `verifyObjectStorageExists` and if there isn't, refactor it to be usable in both cases. 

It also seems a tad weird to me that we have 'doesExist' and 'doesNotExist' that do different things? Could this be just implemented as `doesNotExist() { return !doesExist() }` (similar to how bucketExists/bucketDoesNotExist are)? 

Or is it just a naming-does-not-follow-semantics kind of an issue - eg. should `doesObjectStorageExist` actually be something along the lines of `doesObjectStorageMinioInstanceExist`? (not familiar with the semantics, unfortunately)

Reply:
The reason of not using `doesObjectStorageExists` to check the s3 bucket availability is because of a race condition.
 the s3 endpoint is still availability few seconds after the API delete call, that's why we check against the API and not the resource. BTW terraform should always interract with the API not the resources.

Reply:
Ahh, right - well, if we absolutely cant implement some sort of a waiting logic, then I would propose just naming these more clearly to indicate the difference, and maybe adding the reasoning in a comment to the source?